deepdives/sockets/README.md

```diff
@@ -0,0 +1,183 @@
+Two-way network communication - Sockets
+============================
+
+> Cool examples exist everywhere, from [Messenger](https://wwww.messenger.com) to interactive WebGL games that use WebSockets for their controls, as shown [here](http://www.websocket.org/demos/racer/index.html).
+
+# Definition 
+A socket is one endpoint of a two-way communication link between two programs running on the network. A socket is bound to a port number so that the TCP/HTTP layer can identify the application that data is destined to be sent to. 
+
+We have learned so far that a server runs on a machine at a specific port. It is also true that a server has a socket mounted on it that is bound to a specific port number. The server waits, listening to the socket for a client to make a connection request. Schematically, we can vizualize this as shown below. 
```

Comment:
The "It is also true..." sentence:

A socket is an abstraction on top of an address for a particular port on a particular server.